Network: Add support for using bridged NIC IP filtering on unmanaged parent bridges

tomponline opened the pull request:
This requires the user specifies which IP the NIC should be using by setting `ipv{n}.address` on the NIC.

Fixes #7477
Closes #8849

stgraber commented:
@tomponline i386 test is failing, possibly because of the older ip6tables on there (i386 is on 18.04, rest is on 20.04).

tomponline commented:
@stgraber that should fix it. As you said, it was failing on i386 due to it using xtables driver. The issue was that before adding support for unmanaged parent networks, the driver could rely on the managed parent network setup step to create the iptables chain used for NIC specific rules (this was added when we introduced bridge ACLs so that we could control the ordering of rules more carefully and ensure that if we wiped all the network level rules and rebuilt them when an ACL changed that the NIC level rules were still processed in the same order - by keeping them in their own chain).

This chain doesn't get created when using an unmanaged parent network, and so it was failing to start the instance. As there is no possibility of a managed ACL for these NICs, I added support for indicating to the firewall driver if the parent network is managed or not, and in the latter case the NIC specific rules are added directly to the INPUT and FORWARD chains as they were in the past.

tomponline commented:
@stgraber tests have passed now.
